### PR TITLE
Aurora Global Database Setup and other updates

### PR DESCRIPTION
**Pull request by catcharbind**
- Aurora Global database setup across 2 AWS regions
- Setup for both Aurora PostgreSQL and MySQL
- Parameter groups
- Enhanced Monitoring

### CODE REVIEW

File: deploy/main.tf [hunk 1]
```diff
@@ -17,9 +17,13 @@ resource "random_string" "rand4" {
...
 }
 
+######################################
+# Generate Tags
+######################################
+
 module "vpc_label" {
-  source    = "aws-ia/label/aws"
-  version   = "0.0.2"
+  source    = "aws-quickstart/label/aws"
```
Comment (tonynv) [inline deploy/main.tf:25]: nothing should be referencing aws-quickstart (use aws-ia)

File: deploy/main.tf [hunk 2]
```diff
@@ -32,25 +36,44 @@ module "vpc_label" {
 # Create VPC
 ######################################
 
-module "aurora_vpc" {
-  source            = "aws-ia/vpc/aws"
-  version           = "0.0.2"
+module "aurora_vpc_p" {
+  source            = "aws-quickstart/vpc/aws"
```
Comment (tonynv) [inline deploy/main.tf:40]: nothing should be referencing aws-quickstart (use aws-ia)

File: setup_workspace/workspace.tf [hunk 3]
```diff
@@ -10,8 +10,8 @@ resource "random_pet" "name" {
 # Generate new terraform org and workspace
 
 module "tfc_workspace" {
-  source                = "aws-ia/cloud_workspace/hashicorp"
-  version               = "0.0.1"
+  source                = "aws-quickstart/tfc_workspace/aws"
```
Comment (tonynv) [inline setup_workspace/workspace.tf:13]: nothing should be referencing aws-quickstart (use aws-ia)

File: deploy/main.tf [hunk 4]
```diff
@@ -32,25 +36,44 @@ module "vpc_label" {
...
   cidr              = "10.0.0.0/16"
-  public_subnets    = ["10.0.128.0/20", "10.0.144.0/20", "10.0.160.0/20", "10.0.176.0/20", "10.0.240.0/22", "10.0.244.0/22"]
-  private_subnets_A = ["10.0.0.0/19", "10.0.32.0/19", "10.0.64.0/19", "10.0.96.0/19", "10.0.232.0/22", "10.0.236.0/22"]
+  public_subnets    = ["10.0.0.0/20", "10.0.32.0/20", "10.0.64.0/20"]
+  private_subnets_A = ["10.0.16.0/20", "10.0.48.0/20", "10.0.80.0/20"]
+  tags              = module.vpc_label.tags
+}
+
+module "aurora_vpc_s" {
+  source            = "aws-quickstart/vpc/aws"
```
Comment (tonynv) [inline deploy/main.tf:50]: nothing should be referencing aws-quickstart (use aws-ia)

**Review by tonynv — CHANGES_REQUESTED** [review]
Do not reference aws-quickstart, everything should be pointed to aws-ia

File: deploy/main.tf (hunk 1 above)
Comment (catcharbind) [reply deploy/main.tf]: Updated!

**Review by tonynv — COMMENTED** [review]
By default it create the global db.  I suggest we create two entry points in the deploy folder deploy/standard_db and deploy/global_db  the current main should move into that folder. I have also updated the vpc module with create_vpc. With option we could potentialy keep a single entry point under deploy

**Comment by tonynv** [comment]
Resource module.aurora.aws_kms_key.kms_p has lifecycle.prevent_destroy set. This prevent clean deletes

**Comment by catcharbind** [comment]
Updated setup_globaldb and storage_encrypted to false by default. Updated VPC module version to create vpc conditionally.
KMS keys need to be retained for any encrypted Aurora snapshots to be retained.

**Review by tonynv — APPROVED** [review]

**Pull request merged**